cim-common/src/test/java/com/crossoverjie/cim/common/route/algorithm/consistenthash/SortArrayMapConsistentHashTest.java

```diff
@@ -103,12 +112,49 @@ public void getFirstNodeValue7() {
         strings.add("45.78.28.220:9100:10081") ;
         strings.add("45.78.28.220:9100:00081") ;
 
-        String process = map.process(strings,"1551253899106");
+        String PROCESS = map.process(strings,"1551253899106");
+        for (int i = 0; i < 100; i++) {
+            String process = map.process(strings, "1551253899106");
+            Assert.assertEquals(PROCESS, process);
+        }
+    }
+
+    @Test
+    public void getFirstNodeValue8() {
+        AbstractConsistentHash map = new SortArrayMapConsistentHash() ;
 
-        System.out.println(process);
-        Assert.assertEquals("45.78.28.220:9100:00081",process);
+        List<String> strings = new ArrayList<>();
+        for (int i = 0; i < 10; i++) {
+            strings.add("127.0.0." + i);
+        }
+        Set<String> processes = new HashSet<>();
+        for (int i = 0; i < 10; i++) {
+            String process = map.process(strings,"zhangsan" + i);
+            processes.add(process);
+        }
+        RangeCheckTestUtil.assertInRange(processes.size(), 2, 10);
     }
 
+    @Test
+    public void testVirtualNode() throws NoSuchFieldException, IllegalAccessException {
+        AbstractConsistentHash map = new SortArrayMapConsistentHash();
 
+        List<String> strings = new ArrayList<>();
+        for (int i = 0; i < 10; i++) {
+            strings.add("127.0.0." + i);
+        }
+
+        String process = map.process(strings,"zhangsan");
+        Field sortArrayMapField = SortArrayMapConsistentHash.class.getDeclaredField("sortArrayMap");
```

Comment:
There is no need to use reflection here, you can add a method for testing, like this:

```java
    @VisibleForTesting
    public SortArrayMap getSortArrayMap() {
        return sortArrayMap;
    }
```